bench.py

```diff
@@ -0,0 +1,416 @@
+    # This code has explicit conversions and will raise an exception if it doesn't know how to convert.
+    # This code uses the python 3.10 match statement if the python version is 3.10 or greater.
+    ncclAlgorithm = ""
+    #    match mscclAlgorithm.lower():
+    #        case "allreduce":
+    #            ncclAlgorithm = "all_reduce"
+    #        case "allgather":
+    #            ncclAlgorithm = "all_gather"
+    #        case "alltoall":
+    #            ncclAlgorithm = "all_to_all"
+    #        case _:
+    #            raise Exception("Unknown algorithm: " + mscclAlgorithm)
+    if mscclCollective.lower() == "allreduce":
+        ncclAlgorithm = "all_reduce"
+    elif mscclCollective.lower() == "allgather":
+        ncclAlgorithm = "all_gather"
+    elif mscclCollective.lower() == "alltoall":
+        ncclAlgorithm = "all_to_all"
+    else:
+        raise Exception("Unexpected algorithm: " + mscclAlgorithmNamePrefix)
+
+    ncclPerfTest = ncclAlgorithm + "_perf"
+
+    # Construct the command to run the benchmarks using the mpirun command
+    # example command:
+    #    mpirun --bind-to numa --tag-output --allow-run-as-root -np 8  -mca pml ob1 -mca btl ^openib -mca btl_tcp_if_include eth0 -x PATH -x LD_LIBRARY_PATH=~/msccl/build/lib/:$LD_LIBRARY_PATH -x UCX_IB_ENABLE_CUDA_AFFINITY=n -x NCCL_IB_PCI_RELAXED_ORDERING=1 -x UCX_IB_PCI_RELAXED_ORDERING=on -x UCX_NET_DEVICES=mlx5_0:1 -x UCX_TLS=rc -x NCCL_SOCKET_IFNAME=eth0 -x NCCL_DEBUG=INFO -x NCCL_NET_GDR_LEVEL=5 -x NCCL_DEBUG_SUBSYS=INIT,ENV -x NCCL_ALGO=MSCCL,RING,TREE -x CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 -x MSCCL_XML_FILES=/home/saemal/test.xml ~/nccl-tests/build/all_reduce_perf -b 1KB -e 32MB -f 2 -g 1 -c 1 -w 100 -n 100
+    #
+    # below are important components of the command
+    #
+    commandPrefix = "mpirun --bind-to numa --tag-output --allow-run-as-root"
+    numProcs = " -np 16"  # for the -np parameter
+    mcaParams = " -hostfile /job/hostfile -mca pml ob1 -mca btl ^openib -mca btl_tcp_if_include eth0"
+
+    # The environment variables that are set for the command
+    # TODO: Consider separating these out variables that are non-constant. Originally the path prefix
+    #    was ~/ but seemed to have problems, so I changed it to the full path.
+    envVars = " -x PATH -x LD_LIBRARY_PATH=/home/saemal/msccl/build/lib/:$LD_LIBRARY_PATH -x UCX_IB_ENABLE_CUDA_AFFINITY=n -x NCCL_IB_PCI_RELAXED_ORDERING=1 -x UCX_IB_PCI_RELAXED_ORDERING=on -x UCX_NET_DEVICES=mlx5_0:1 -x UCX_TLS=rc -x NCCL_SOCKET_IFNAME=eth0 -x NCCL_DEBUG=INFO -x NCCL_NET_GDR_LEVEL=5 -x NCCL_DEBUG_SUBSYS=INIT,ENV -x NCCL_ALGO=MSCCL,RING,TREE -x CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7"
+
+    mscclEnvVars = " -x MSCCL_XML_FILES=" + os.path.join(inputDirectory, filename)
+    ncclBufferRange = f" -b 1KB -e {endBufSize}"
+    ncclParameters = " /home/saemal/msccl-tools/msccl/autosynth/msccl_ndv2_launcher.sh /home/saemal/nccl-tests/build/" + ncclPerfTest + ncclBufferRange + " -f 2 -g 1 -c 1 -w 100 -n 100"
```

Comment:
yeah all of these could be cleaned up if we take as input an example from the user.

Reply:
Done